internal/subscription/core.go

```diff
@@ -343,54 +342,39 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
 
 	logger.Ctx(ctx).Infow("migration: found subscriptions to migrate", "count", len(subscriptionsToUpdate))
 
-	topicNames := make([]string, 0)
-	for _, model := range subscriptionsToUpdate {
-		needsUpdate := false
-
-		// update retry policy if not set
-		if model.RetryPolicy == nil {
-			model.setDefaultRetryPolicy()
-			// update dead letter policy every time as previously we were saving only dead-letter topic names
-			// this takes care of updating max_delivery_attempts
-			model.setDefaultDeadLetterPolicy()
-			needsUpdate = true
+	updatedSubCount := 0
+	for _, sub := range subscriptionsToUpdate {
+		if sub.PushConfig == nil {
+			continue
 		}
-
-		if needsUpdate {
-			// collect all the delay topic names to be created
-			topicNames = append(topicNames, model.GetDelayTopics()...)
-
-			logger.Ctx(ctx).Infow("migration: updating subscription", "model", model.Name)
-			// update the subscription model
-			err := c.UpdateSubscription(ctx, model)
-			if err != nil {
-				return err
+		pushAttr := sub.PushConfig.Attributes
+		if pushAttr != nil {
+			needsUpdate := false
+			if _, ok := pushAttr[attributeUsername]; ok {
+				logger.Ctx(ctx).Infow("migration: deleting username", "username", pushAttr[attributeUsername], "subscription", sub.Name)
+				delete(pushAttr, attributeUsername)
+				needsUpdate = true
 			}
-		}
-	}
 
-	logger.Ctx(ctx).Infow("migration: delay topics to create", "count", len(topicNames), "topicNames", topicNames)
-	// collect the success and failed topic names so that they can be retried if needed
-	successTopicNames, failedTopicNames := make([]string, 0), make([]string, 0)
-	for _, tName := range topicNames {
-		tModel, terr := topic.GetValidatedModel(ctx, &metrov1.Topic{
-			Name: tName,
-		})
-		if terr != nil {
-			logger.Ctx(ctx).Errorw("migration: failed to create validated topic model", "tName", tName, "error", terr.Error())
-			failedTopicNames = append(failedTopicNames, tName)
-			continue
-		}
+			if _, ok := pushAttr[attributePassword]; ok {
+				logger.Ctx(ctx).Infow("migration: deleting password", "password", pushAttr[attributePassword], "subscription", sub.Name)
```

Comment:
Don't log. `password` is sensitive info. The log can say `deleting password for username` instead